features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -93,17 +96,51 @@ internal class Base64Serializer private constructor(
         imageWireframe: MobileSegment.Wireframe.ImageWireframe,
         base64SerializerCallback: Base64SerializerCallback?
     ) {
-        val base64String = convertBmpToBase64(drawable, bitmap, shouldCacheBitmap)
-        finalizeRecordedDataItem(base64String, imageWireframe, base64SerializerCallback)
+        val byteArray = compressBitmap(bitmap)
+        val base64String = convertBmpToBase64(byteArray, drawable, bitmap, shouldCacheBitmap)
+        val hash = calculateHash(drawable, byteArray)
+        finalizeRecordedDataItem(base64String, hash, imageWireframe, base64SerializerCallback)
+    }
+
+    private fun calculateHash(
```

Comment:
There's already an SHA256 hash generator in the core, so it could be better to add an MD5 generator there and then use it here